From ded8280b498521f193e9486f78689d6d5e1ae809 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Tue, 31 Mar 2020 19:34:20 +0800
Subject: [PATCH] video: rockchip: mpp: fix issue about setting hevc cabac
 clock

1. add video_hevc_cabac to clock list
2. set default hevc cabac clock
3. set clock for each task

Change-Id: Ic09dcc2267d8ca7c745fdbc70f079de06fde90e9
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_common.c | 1 +
 drivers/video/rockchip/mpp/mpp_rkvdec.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/drivers/video/rockchip/mpp/mpp_common.c b/drivers/video/rockchip/mpp/mpp_common.c
index 3f6893d0c3c8..d8cfc04c7552 100644
--- a/drivers/video/rockchip/mpp/mpp_common.c
+++ b/drivers/video/rockchip/mpp/mpp_common.c
@@ -314,6 +314,7 @@ mpp_reset_control_get(struct mpp_dev *mpp, const char *name)
 		"video_a", "video_h",
 		"niu_a", "niu_h",
 		"video_cabac", "video_core",
+		"video_hevc_cabac",
 	};
 
 	if (!name) {
diff --git a/drivers/video/rockchip/mpp/mpp_rkvdec.c b/drivers/video/rockchip/mpp/mpp_rkvdec.c
index b4aa0743e95e..7c03d66f920c 100644
--- a/drivers/video/rockchip/mpp/mpp_rkvdec.c
+++ b/drivers/video/rockchip/mpp/mpp_rkvdec.c
@@ -1468,6 +1468,8 @@ static int rkvdec_set_freq(struct mpp_dev *mpp,
 		clk_set_rate(dec->aclk, task->aclk_freq * MHZ);
 		clk_set_rate(dec->clk_cabac, task->clk_cabac_freq * MHZ);
 		clk_set_rate(dec->clk_core, task->clk_core_freq * MHZ);
+		clk_set_rate(dec->clk_hevc_cabac,
+			     task->clk_hevc_cabac_freq * MHZ);
 	}
 
 	return 0;
-- 
2.35.3

